src/authentication/EmptyCredentialsExtractor.ts

```diff
@@ -8,13 +6,6 @@ import { CredentialsExtractor } from './CredentialsExtractor';
  * Extracts the empty credentials, indicating an unauthenticated agent.
  */
 export class EmptyCredentialsExtractor extends CredentialsExtractor {
-  public async canHandle({ headers }: HttpRequest): Promise<void> {
-    const { authorization } = headers;
-    if (authorization) {
-      throw new NotImplementedHttpError('Unexpected Authorization scheme.');
-    }
-  }
-
```

Comment:
Are we sure, and why?

Perhaps update top comment to reflect this.

Reply:
Right, should have added a reason somewhere. Problem is that it's related to a completely different component.

Without this there was a problem during the OIDC conversation. At some point the client sends a request with an auth header specifically for OIDC. This caused all our CredentialsExtractors to fail since none of them supported it (as they should), but also this one since there was still an auth header. Result being that no allowed permissions were generated (since there were no credentials to generate them for), causing the authorizer to reject that request. (That was fun to debug 😉 ).

Reply:
Check, thanks!